lib/rubocop/cop/sorbet/obsolete_strict_memoization.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require "rubocop"
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # This cop checks for the obsolete pattern for initializing instance variables that was required for older Sorbet
+      # versions in `#typed: strict` files.
+      #
+      # It's no longer required, as of Sorbet 0.5.10210
+      # See https://github.com/sorbet/sorbet/issues/995
+      #
+      # @example
+      #
+      #   # bad
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo = T.let(@foo, T.nilable(Foo))
+      #     @foo ||= Foo.new
+      #   end
+      #
+      #   # good
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo ||= T.let(Foo.new, T.nilable(Foo))
+      #   end
+      #
+      # TODO: find the right way to access the line length limit, indentation style, and indentation width.
+      class ObsoleteStrictMemoization < RuboCop::Cop::Base
+        include RuboCop::Cop::MatchRange
+        extend AutoCorrector
+
+        include TargetSorbetVersion
+        minimum_target_sorbet_static_version "0.5.10210"
```

Comment:
> Should this be a constant?

I don't think so, it would just add indirection without any novel information, e.g.

```suggestion
        MINIMUM_SUPPORTED_VERSION = "0.5.10210"
        minimum_target_sorbet_static_version MINIMUM_SUPPORTED_VERSION
```

or did you mean something else?

> If we expect it to be modified

We don't, this is a quality of the cop (and Sorbet's support for the syntax it corrects to), which wouldn't be user configurable

Reply:
No, I mean to not use `minimum_target_sorbet_static_version` at all and just a constant:

```rb
MINIMUM_TARGET_SORBET_STATIC_VERSION = Gem::Version.new(version)
```

And then in the helper method:

```rb
def support_target_sorbet_static_version?(version)
  MINIMUM_TARGET_SORBET_STATIC_VERSION<= Gem::Version.new(version)
end
```

Reply:
Ah, I see what you mean.

I wanted it to be a declarative style, similar to `minimum_target_rails_version` https://github.com/search?q=repo%3Arubocop%2Frubocop-rails+minimum_target_rails_version&type=code

See https://github.com/rubocop/rubocop/issues/11920 for context